src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/backpacks/SlimefunBackpack.java

```diff
@@ -65,7 +66,13 @@ public boolean isItemAllowed(@Nonnull ItemStack item, @Nullable SlimefunItem ite
         if (SlimefunTag.SHULKER_BOXES.isTagged(item.getType())) {
             return false;
         }
-
+        if (this instanceof HazmatBackpack && !(itemAsSlimefunItem instanceof Radioactive)) {
+            return false;
+        }
+        if (itemAsSlimefunItem instanceof Radioactive && !(this instanceof HazmatBackpack)) {
+            return false;
+        }
```

Comment:
This is still super redundant and should belong into the HazmatBackpack class...

Reply:
Well one of them prevents the radioactive item from being entered into the normal backpacks, if i put that in the HazmatBackpack wouldn't that not trigger on normal backpacks?
Also since I'd be overriding the method from here, i will need to do both checks again anyway.
Or am I trippin?

Reply:
Look into polymorphism.

You can override a method in child classes to replace the behaviour or fallback to the parent behaviour.
It is a very bad design pattern to do instanceof checks for child classes inside the parent class, as it renders the child class useless and goes against the principles of polymorphism.

Think about it simply like this:

What does the normal Backpack do? Allow every item except for backpacks, shulkers and radioactive materials.
The method should reflect this.

What exception does the Hazmat Backpack provide? Radioactive substances ARE allowed.
So the method **inside** HazmatBackpack should have a special case for that and fallback to the default behaviour otherwise. Since the radioactive check will no longer pass due to the prior exclusion, the instanceof check is made obsolete.